src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/FileWatchedPortableExecutableReferenceFactory.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.MetadataReferences
+{
+    [Export]
+    internal sealed class FileWatchedPortableExecutableReferenceFactory
+    {
+        private readonly object _gate = new object();
+
+        /// <summary>
+        /// This right now acquires the entire VisualStudioWorkspace because right now the production
+        /// of metadata references depends on other workspace services. See the comments on
+        /// <see cref="VisualStudioMetadataReferenceManagerFactory"/> that this strictly shouldn't be necessary
+        /// but for now is quite the tangle to fix.
```

Comment:
>but for now is quite the tangle to fix. [](start = 12, length = 39)

Do we have an issue that tracks untangling it?

Reply:
Ah, thought we had one but didn't. https://devdiv.visualstudio.com/DevDiv/_workitems/edit/777154 filed.